app/controllers/materials_controller.rb

```diff
@@ -0,0 +1,51 @@
+class MaterialsController < ApplicationController
+
+  def index
+    @materials = policy_scope(Material).all
+  end
+
+  def show
+    @material = Material.find(params[:id])
+    authorize @material
+  end
+
+  def new
+    @material = Material.new
+    authorize @material
+  end
+
+  def create
+    @material = Material.new(material_params)
+    if @material.save
+      redirect_to materials_path
+    else
+      render :new
+    end
+    authorize @material
+  end
+
+  def edit
+    @material = Material.find(params[:id])
+    authorize @material
+  end
+
+  def update
+    @material = Material.find(params[:id])
+    @material.update(material_params)
+    authorize @material
+    redirect_to material_path(@material)
+  end
+
+  def destroy
+    @material = Material.find(params[:id])
+    @material.destroy
+    authorize @material
```

Comment:
What is the interets to authorize this @ material if destroyed ?

Reply:
@Nicode80 the interest is that you to authorize the variable for pundit to check the policy. Pundit need to check the following condition: `record.user == user` (in Pundit user = current_user and record is, in this case @material, that you need to authorize for Pundit to access it). The convention is to place authorize at the bottom of your method.